tests/components/songpal/test_media_player.py

```diff
@@ -0,0 +1,287 @@
+"""Test songpal media_player."""
+from datetime import timedelta
+
+from songpal import (
+    ConnectChange,
+    ContentChange,
+    PowerChange,
+    SongpalException,
+    VolumeChange,
+)
+
+from homeassistant.components import media_player, songpal
+from homeassistant.components.songpal.const import SET_SOUND_SETTING
+from homeassistant.components.songpal.media_player import (
+    STATE_OFF,
+    STATE_ON,
+    SUPPORT_SONGPAL,
+)
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.setup import async_setup_component
+from homeassistant.util import dt as dt_util
+
+from . import (
+    CONF_DATA,
+    CONF_ENDPOINT,
+    CONF_NAME,
+    ENDPOINT,
+    ENTITY_ID,
+    FRIENDLY_NAME,
+    MAC,
+    MODEL,
+    SW_VERSION,
+    _create_mocked_device,
+    _patch_media_player_device,
+)
+
+from tests.async_mock import AsyncMock, MagicMock, call, patch
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+def _get_attributes(hass):
+    state = hass.states.get(ENTITY_ID)
+    return state.as_dict()["attributes"]
+
+
+async def test_setup_platform(hass):
+    """Test the legacy setup platform."""
+    mocked_device = _create_mocked_device(throw_exception=True)
+    with _patch_media_player_device(mocked_device):
+        await async_setup_component(
+            hass,
+            media_player.DOMAIN,
+            {
+                media_player.DOMAIN: [
+                    {
+                        "platform": songpal.DOMAIN,
+                        CONF_NAME: FRIENDLY_NAME,
+                        CONF_ENDPOINT: ENDPOINT,
+                    }
+                ],
+            },
+        )
+        await hass.async_block_till_done()
+
+    # No device is set up
+    mocked_device.assert_not_called()
+    all_states = hass.states.async_all()
+    assert len(all_states) == 0
+
+
+async def test_setup_failed(hass):
+    """Test failed to set up the entity."""
+    mocked_device = _create_mocked_device(throw_exception=True)
+    entry = MockConfigEntry(domain=songpal.DOMAIN, data=CONF_DATA)
+    entry.add_to_hass(hass)
+
+    logger_warning = MagicMock()
+    logger_error = MagicMock()
+    with patch("logging.Logger.warning", side_effect=logger_warning), patch(
```

Comment:
There's a `caplog` fixture available to assert logs:
https://docs.pytest.org/en/latest/logging.html#caplog-fixture